src/validate-params.ts

```diff
@@ -2,22 +2,22 @@ import {joinErrors, toResult, Result} from './util';
 import * as chalk from 'chalk';
 import SoyContext from './soy-context';
 import JSContext from './js-context';
+import {Config} from './config';
 
-export default function validateParams(soyContext: SoyContext, jsContext: JSContext): Result {
-  /**
-   * We just skip validation if the default class does not extend from Component.
-   * TODO: See if we can resolve and parse the parent class' STATE.
-   */
-  if (jsContext.getSuperClassImportPath() !== 'metal-component') {
-    return toResult(true);
-  }
+function getImplicitParams(className: string | null, paramsMap: any):string[] {
```

Comment:
So I've been preferring `Array<string>` over `string[]` because I've been using tuples in some places, and it's also more consistent with the type annotations for other ADT's.